app/src/main/java/de/danoeh/antennapod/activity/PlaybackSpeedDialogActivity.java

```diff
@@ -0,0 +1,26 @@
+package de.danoeh.antennapod.activity;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.os.Bundle;
+
+import de.danoeh.antennapod.core.preferences.UserPreferences;
+import de.danoeh.antennapod.dialog.VariableSpeedDialog;
+
+public class PlaybackSpeedDialogActivity extends AppCompatActivity {
+
+    VariableSpeedDialog speedDialog = new VariableSpeedDialog();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        setTheme(UserPreferences.getTranslucentDialogTheme());
+        super.onCreate(savedInstanceState);
+        speedDialog.show(getSupportFragmentManager(), null);
+    }
+
+    @Override
+    public void onBackPressed() {
+        super.onBackPressed();
+        finish();
+    }
```

Comment:
When you overwrite `VariableSpeedDialog` (right here in the Activity), you can finish the activity when the dialog is cancelled. Then you do not need to "work around" the Android back button behavior.

Reply:
Yes, I've deleted this line of code, it really doesn't help in this case.